lib/rules/syntax-string-no-invalid/index.mjs

```diff
@@ -0,0 +1,80 @@
+import { isTokenString, tokenize } from '@csstools/css-tokenizer';
+import { fork } from 'css-tree';
+
+import { atRuleRegexes } from '../../utils/regexes.mjs';
+import { declarationValueIndex } from '../../utils/nodeFieldIndices.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'syntax-string-no-invalid';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (syntaxString) => `Unexpected invalid syntax string "${syntaxString}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/syntax-string-no-invalid',
+};
+
+const forkedLexer = fork({
+	properties: {
+		'-stylelint-syntax': "'*' | <syntax-component> [ <syntax-combinator> <syntax-component> ]*",
+	},
+	types: {
+		'syntax-component': "<syntax-single-component> <syntax-multiplier>? | '<' transform-list '>'",
+		'syntax-single-component': "'<' <syntax-type-name> '>' | <ident>",
+		'syntax-type-name':
+			'angle | color | custom-ident | image | integer | length | length-percentage | number | percentage | resolution | string | time | url | transform-function',
+		'syntax-combinator': "'|'",
+		'syntax-multiplier': "[ '#' | '+' ]",
+	},
+}).lexer;
```

Comment:
I also had it placed here in my example, but maybe better to run this after `if (!validOptions)`?

Then this isn't computed unless it will be used.

Reply:
Good suggestion! Fixed in this commit.
[Refactor adjust the position of lexer fork](https://github.com/stylelint/stylelint/pull/8331/commits/ec664561015b2e9b3590bc144a0fab3c3a87244f)

Reply:
Thank you